src/decorators/authorizationDecorator.js

```diff
@@ -10,29 +10,37 @@
  *  *******************************************************************************
  *
  */
-const logger = require('../logger/index');
+const logger = require('../logger');
+const config = require('../config');
 const UserManager = require('../sequelize/managers/userManager');
+const AccessTokenManager = require('../sequelize/managers/accessTokenManager');
+const Errors = require('../utils/errors');
 
-function checkUserExistance(f) {
+function checkAuthorization(f) {
   return async function() {
 
-    logger.info('checking user token: ' + token);
+    const fArgs = Array.prototype.slice.call(arguments);
+    const req = fArgs[0];
+    const token = req.headers.authorization;
 
-    let fArgs = Array.prototype.slice.call(arguments);
-    let req = fArgs[0];
-    let token = req.headers.authorization;
+    logger.info('checking user token: ' + token);
 
-    let user = await UserManager.findByToken(token);
+    const user = await UserManager.findByAccessToken(token);
 
-    if (user) {
-      return await f.apply(this, arguments);
-    } else {
+    if (!user) {
       logger.error('token ' + token + ' incorrect');
-      throw 'authorization failed'
+      throw new Errors.AuthorizationError('authorization failed');
```

Comment:
this should be `AuthenticationError`, sorry if my comment confused you!
`Authorization` errors should be thrown when an authenticated user tries to access other users' resources. like `user1` has a valid token, but calls the API to modify a microservice that belongs to `user2`.